src/oss-find-squats-lib/MutateExtension.cs

```diff
@@ -104,32 +104,54 @@ public static class MutateExtension
 
             Dictionary<string, IList<Mutation>> generatedMutations = new();
 
-            // Check to see if it is a scoped npm package to generate candidates for.
-            bool isScoped = purl.HasNamespace();
-            string nameToMutate = isScoped ? purl.Namespace : purl.Name;
+            // Check to see if the package has a namespace to generate candidates for.
+            bool hasNamespace = purl.HasNamespace();
```

Comment:
This might be my fault based on the existing code, but this logic is actually very confusing and I think it could be improved just with an if(hasNamespace) splitting the two flows earlier.

Reply:
I tried to clean up the surrounding logic best I could. Let me know what you think.

Reply:
Much cleaner.

Reply:
Managed to move this to the new Generate(PackageUrl) implementation, which I think is much cleaner.